CMakeLists.txt

```diff
@@ -109,16 +109,28 @@ else()
 
   # No config file for this
   if(WITH_GFLAGS)
-    find_package(gflags REQUIRED)
+    option(GFLAGS_USE_TARGET_NAMESPACE "Use gflags import target with namespace." ON)
+    find_package(gflags CONFIG)
+    if(gflags_FOUND)
+      list(APPEND THIRDPARTY_LIBS ${GFLAGS_TARGET})
+    else()
+      find_package(gflags REQUIRED)
+      list(APPEND THIRDPARTY_LIBS gflags::gflags)
+      include_directories(${gflags_INCLUDE_DIR})
+    endif()
     add_definitions(-DGFLAGS=1)
-    include_directories(${gflags_INCLUDE_DIR})
-    list(APPEND THIRDPARTY_LIBS gflags::gflags)
   endif()
 
   if(WITH_SNAPPY)
-    find_package(snappy REQUIRED)
+    find_package(Snappy CONFIG)
+    if(Snappy_FOUND)
+      list(APPEND THIRDPARTY_LIBS Snappy::snappy)
+    else()
+      message("Snappy config not found. Fallback to legacy find.")
+      find_package(snappy REQUIRED)
```

Comment:
could you rename `cmake/modules/Findsnappy.cmake` to `cmake/modules/FindSnappy.cmake` and also apply following patch?
```diff
diff --git a/cmake/modules/Findsnappy.cmake b/cmake/modules/Findsnappy.cmake
index 6ed5fda3d..cb59b5e49 100644
--- a/cmake/modules/Findsnappy.cmake
+++ b/cmake/modules/Findsnappy.cmake
@@ -14,7 +14,7 @@ find_library(SNAPPY_LIBRARIES
   HINTS ${SNAPPY_ROOT_DIR}/lib)
 
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(snappy DEFAULT_MSG SNAPPY_LIBRARIES SNAPPY_INCLUDE_DIR)
+find_package_handle_standard_args(Snappy DEFAULT_MSG SNAPPY_LIBRARIES SNAPPY_INCLUDE_DIR)
 
 mark_as_advanced(
   SNAPPY_LIBRARIES
```

Reply:
Nice! As long as you're OK with that~